packages/kokkos-kernels/src/sparse/impl/KokkosSparse_spgemm_impl_kkmem.hpp

```diff
@@ -492,7 +412,7 @@ struct KokkosSPGEMM
 
     hm2.keys = (nnz_lno_t *) (tmp);
     tmp += max_nnz;
-    hm2.values = (scalar_t *) (tmp);
+    hm2.values = KokkosKernels::Impl::alignPtr<volatile nnz_lno_t*, scalar_t>(tmp);
```

Comment:
Do volatile pointers have different alignment than non-volatile pointers?

Reply:
Not that I know of, but tmp is volatile so I didn't want to take chances with cv- conversion warnings or errors. I don't actually think the 'volatile' here is necessary, but that's a different issue.

Reply:
Like the macro you saw above, SPGEMM has a lot of little cleanup things that need to be done. Another one is that $view.data() gets taken as a raw pointer for the numeric functors, instead of just storing and accessing the view.